### PR TITLE
Migrates todo-mvp to androidx

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: todoapp/app/build.gradle [hunk 1]
```diff
@@ -65,45 +63,46 @@ android {
...
-    compile "com.google.guava:guava:$rootProject.guavaVersion"
-    compile "android.arch.persistence.room:runtime:$rootProject.roomVersion"
-    annotationProcessor "android.arch.persistence.room:compiler:$rootProject.roomVersion"
+    // App"s dependencies, including test
+    implementation "androidx.appcompat:appcompat:$androidXVersion"
+    implementation "androidx.cardview:cardview:$cardVersion"
+    implementation "com.google.android.material:material:$materialVersion"
+    implementation "androidx.recyclerview:recyclerview:$recyclerViewVersion"
+
+    implementation "androidx.legacy:legacy-support-v4:1.0.0"
```
Comment (florina-muntenescu) [inline todoapp/app/build.gradle]: do we need to hardcode this version?

File: todoapp/app/build.gradle [hunk 2]
```diff
@@ -65,45 +63,46 @@ android {
...
-    testCompile "org.hamcrest:hamcrest-all:$rootProject.ext.hamcrestVersion"
+    testImplementation "junit:junit:$rootProject.ext.junitVersion"
+    testImplementation "org.mockito:mockito-all:$rootProject.ext.mockitoVersion"
+    testImplementation "org.hamcrest:hamcrest-all:$rootProject.ext.hamcrestVersion"
 
-    // Android Testing Support Library's runner and rules
-    androidTestCompile "com.android.support.test:runner:$rootProject.ext.runnerVersion"
-    androidTestCompile "com.android.support.test:rules:$rootProject.ext.rulesVersion"
+    // Android Testing Support Library"s runner and rules
+    androidTestImplementation "androidx.test:runner:1.1.1"
```
Comment (florina-muntenescu) [inline todoapp/app/build.gradle]: move these versions to the list of version variables

File: todoapp/app/build.gradle [hunk 3]
```diff
@@ -65,45 +63,46 @@ android {
...
+    androidTestImplementation "androidx.room:room-testing:$roomVersion"
 
     // Dependencies for Android unit tests
-    androidTestCompile "junit:junit:$rootProject.ext.junitVersion"
-    androidTestCompile "org.mockito:mockito-core:$rootProject.ext.mockitoVersion"
-    androidTestCompile 'com.google.dexmaker:dexmaker:1.2'
-    androidTestCompile 'com.google.dexmaker:dexmaker-mockito:1.2'
+    androidTestImplementation "junit:junit:$rootProject.ext.junitVersion"
+    androidTestImplementation "org.mockito:mockito-core:$rootProject.ext.mockitoVersion"
+    androidTestImplementation "com.google.dexmaker:dexmaker:1.2"
```
Comment (florina-muntenescu) [inline todoapp/app/build.gradle]: move to version variables.

Comment (florina-muntenescu) [reply todoapp/app/build.gradle]: and below

File: todoapp/app/proguard-rules.pro [hunk 4]
```diff
@@ -1,10 +1,18 @@
...
--keep class android.support.test.espresso.IdlingRegistry { *; }
+-keep class androidx.drawerlayout.widget.DrawerLayout { *; }
+-keep class androidx.test.espresso.**
+# keep the class and specified members from being removed or renamed
+-keep class androidx.test.espresso.IdlingRegistry { *; }
+-keep class androidx.test.espresso.IdlingResource { *; }
+
 -keep class com.google.common.base.Preconditions { *; }
+
+-keep class androidx.room.RoomDataBase { *; }
```
Comment (florina-muntenescu) [inline todoapp/app/proguard-rules.pro]: would be great if we could add a comment explaining why each of these needs to be kept. 
can be done in a future PR.

File: todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksScreenTest.java [hunk 5]
```diff
@@ -90,11 +94,30 @@
...
                     super.beforeActivityLaunched();
                     // Doing this in @Before generates a race condition.
-                    Injection.provideTasksRepository(InstrumentationRegistry.getTargetContext())
+                    Injection.provideTasksRepository(ApplicationProvider.getApplicationContext())
                         .deleteAllTasks();
                 }
             };
 
+    /**
+     * Prepare your test fixture for this test. In this case we register an IdlingResources with
```
Comment (florina-muntenescu) [inline todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksScreenTest.java]: what's a text fixture?

File: todoapp/build.gradle [hunk 6]
```diff
@@ -27,19 +27,23 @@ ext {
...
-    targetSdkVersion = 26
-    compileSdkVersion = 26
-    buildToolsVersion = '26.0.2'
+    targetSdkVersion = 28
+    compileSdkVersion = 28
 
     // App dependencies
-    supportLibraryVersion = '26.1.0'
-    guavaVersion = '18.0'
+    androidXVersion = '1.1.0-alpha01'
```
Comment (florina-muntenescu) [inline todoapp/build.gradle]: I thought we agreed on using stable versions

**Comment by JoseAlcerreca** [comment]
(Ignore flakiness)

File: todoapp/build.gradle [hunk 7]
```diff
@@ -27,19 +27,26 @@ ext {
...
     mockitoVersion = '1.10.19'
-    powerMockito = '1.6.2'
+    dexMaker = '1.2'
     hamcrestVersion = '1.3'
     runnerVersion = '1.0.1'
     rulesVersion = '1.0.1'
-    espressoVersion = '3.0.1'
-    roomVersion = "1.0.0"
+    espressoVersion = '3.1.1'
+    roomVersion = "2.0.0"
```
Comment (keyboardsurfer) [inline todoapp/build.gradle]: Nit: Keep it consistent and replace " with '

File: todoapp/app/proguard-rules.pro [hunk 8]
```diff
@@ -1,10 +1,18 @@
+-dontoptimize
+
 # Some methods are only called from tests, so make sure the shrinker keeps them.
 -keep class com.example.android.architecture.blueprints.** { *; }
 
--keep class android.support.v4.widget.DrawerLayout { *; }
--keep class android.support.test.espresso.IdlingResource { *; }
--keep class android.support.test.espresso.IdlingRegistry { *; }
+-keep class androidx.drawerlayout.widget.DrawerLayout { *; }
```
Comment (nickbutcher) [inline todoapp/app/proguard-rules.pro]: This shouldn't be needed as AAPT emits keep rules for Views?!

File: todoapp/app/src/main/res/layout/addtask_act.xml [hunk 9]
```diff
@@ -14,34 +14,34 @@
...
   -->
 
-<android.support.v4.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:id="@+id/drawer_layout"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context=".addedittask.AddEditTaskActivity">
+<androidx.drawerlayout.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (nickbutcher) [inline todoapp/app/src/main/res/layout/addtask_act.xml]: Strange formatting?

File: todoapp/app/src/main/res/layout/statistics_act.xml [hunk 10]
```diff
@@ -14,33 +14,33 @@
...
 
-<android.support.v4.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:id="@+id/drawer_layout"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context=".tasks.TasksActivity"
-    tools:openDrawer="start">
+<androidx.drawerlayout.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (nickbutcher) [inline todoapp/app/src/main/res/layout/statistics_act.xml]: Strange formatting?

File: todoapp/app/src/main/res/layout/tasks_act.xml [hunk 11]
```diff
@@ -14,35 +14,35 @@
...
 
-<android.support.v4.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:id="@+id/drawer_layout"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context=".tasks.TasksActivity"
-    tools:openDrawer="start">
+<androidx.drawerlayout.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (nickbutcher) [inline todoapp/app/src/main/res/layout/tasks_act.xml]: Ditto

File: todoapp/app/src/main/res/layout/addtask_act.xml (hunk 9 above)
Comment (JoseAlcerreca) [reply todoapp/app/src/main/res/layout/addtask_act.xml]: fixed

File: todoapp/app/proguard-rules.pro (hunk 8 above)
Comment (JoseAlcerreca) [reply todoapp/app/proguard-rules.pro]: java.lang.NoSuchMethodError: No virtual method isDrawerVisible(I)Z in class Landroidx/drawerlayout/widget/DrawerLayout; or its super classes (declaration of 'androidx.drawerlayout.widget.DrawerLayout' appears in /data/app/com.example.android.architecture.blueprints.todomvp.mock-0ZkU6c_Gfa12K-fi5IOT1A==/base.apk)

File: todoapp/app/src/main/res/layout/tasks_act.xml (hunk 11 above)
Comment (JoseAlcerreca) [reply todoapp/app/src/main/res/layout/tasks_act.xml]: Fixed all xmls

File: todoapp/app/src/main/res/layout/statistics_act.xml (hunk 10 above)
Comment (JoseAlcerreca) [reply todoapp/app/src/main/res/layout/statistics_act.xml]: Fixed

**Comment by JoseAlcerreca** [comment]
Thanks, all comments addressed, PTAL.

**Pull request merged**